ext/openssl/openssl.c

```diff
@@ -857,8 +864,8 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 			to_add_len = ASN1_STRING_to_UTF8(&to_add, str);
 			needs_free = 1;
 		} else {
-			/* ASN1_STRING_data(3): Since this is an internal pointer it should not be freed or modified in any way */
-			to_add = ASN1_STRING_data(str);
+			/* ASN1_STRING_get0_data(3): Since this is an internal pointer it should not be freed or modified in any way */
+			to_add = (char *)ASN1_STRING_get0_data(str);
```

Comment:
That's not good. Discarting const qualifier by casting is something that we should eliminate. I see why it's done but in this case it's better to have copy of data rather than discard const.